wandb/sdk/internal/sender.py

```diff
@@ -880,6 +880,25 @@ def _sync_spell(self) -> None:
             pass
         # TODO: do something if sync spell is not successful?
 
+    def _setup_fork(self):
+        if self._settings.fork_from:
+            first_step = int(self._settings.fork_from.value) + 1
+            self._resume_state.step = first_step
+
+            # at this point, we have already created the run. now we need
+            # to query the newly created forked run to get the history
+            # line count. this information is calculated on the backend
+            # and installed on the new run metadata object.
+            fork_state = self._api.run_resume_status(
+                entity=self._run.entity,  # type: ignore
+                project_name=self._run.project,
+                name=self._run.run_id,
```

Comment:
i refactored this to have upsertBucket return historyLineCount and save an API call. now we don't need to call run_resume_status here -- we just get it from the upsertBucket response produced by _init_run(). 

we have to upsert the run to get the correct history line count for the forked run and set the filestream offsets appropriately. because the backend does this during the execution of the upsertBucket mutation we need to wait until that happens to read off the right historyLineCount for the new forked run and set it in the client so that we log appropriately to filestream. 

the fact that it is set on the new run metadata object (in the metadata store) is relevant because we need to query that store to get this information (which we do here).

Reply:
Thanks, that's better!